### PR TITLE
Spectral library download added to FileNamesParameter

### PR DESCRIPTION
**Pull request by robinschmid**
Now also captures multi file download progress and existing files

image

image

Relies on #2120 done

### CODE REVIEW

File: mzmine-community/src/main/java/io/github/mzmine/modules/io/import_spectral_library/SpectralLibraryImportParameters.java [hunk 1]
```diff
@@ -25,17 +25,20 @@
...
 
 
-  public static final FileNamesParameter dataBaseFiles = new FileNamesParameter(
+  public static final FileNamesWithDownloadParameter dataBaseFiles = new FileNamesWithDownloadParameter(
       "Spectral library files", """
-      Name of file that contains information for peak identification
-      (GNPS json, MONA json, NIST msp, mgf, JCAMP-DX jdx)""", ExtensionFilters.ALL_LIBRARY);
+      Path of spectral library files in common formats
+      (GNPS json, MONA json, NIST msp, mgf, JCAMP-DX jdx)""", ExtensionFilters.ALL_LIBRARY,
+      DownloadAssets.forAssetGroup(AssetGroup.SPECTRAL_LIBRARIES));
```
Comment (SteffenHeu) [inline mzmine-community/src/main/java/io/github/mzmine/modules/io/import_spectral_library/SpectralLibraryImportParameters.java:41]: if we add a download here, we need to conform to CC-BY (mona). for gnps and massbank eu, i cannot find a license,so not sure if we may provide a direct download.

**Comment by SteffenHeu** [comment]
Can you merge the latest master?

**Comment by robinschmid** [comment]
yes

**Comment by robinschmid** [comment]
I will add the better menu item now

**Comment by robinschmid** [comment]
@SteffenHeu done
I will create an issue with your idea to add a last files button also to FileNamesParameter. Dont want to put it into the same PR

File: mzmine-community/src/main/resources/themes/jabref_light.css [hunk 2]
```diff
@@ -566,6 +566,10 @@ TextFlow > .tooltip-text-monospaced {
     -fx-padding: 0.5em 0.41777em 0.5em 0.41777em;
 }
 
+.custom-menu-item {
```
Comment (SteffenHeu) [inline mzmine-community/src/main/resources/themes/jabref_light.css:569]: we put all the additionally added stuff under jabref_additions_common

**Comment by SteffenHeu** [comment]
Is it intended that the files are downloaded every time you click on the libraries? If they are downloaded and present already, i think theyx should just be added.

File: mzmine-community/src/main/java/io/github/mzmine/modules/io/download/DownloadAssetButton.java [hunk 3]
```diff
@@ -66,18 +73,22 @@ public class DownloadAssetButton extends HBox {
...
+  private final ObjectProperty<FileDownloadTask> task = new SimpleObjectProperty<>();
 
 
-  public DownloadAssetButton(@NotNull final ExternalAsset asset,
+  public DownloadAssetButton(@NotNull final List<DownloadAsset> assets) {
+    this(null, assets);
+  }
+
+  public DownloadAssetButton(@Nullable final ExternalAsset asset,
       @NotNull final List<DownloadAsset> assets) {
```
Comment (SteffenHeu) [inline mzmine-community/src/main/java/io/github/mzmine/modules/io/download/DownloadAssetButton.java:85]: What's the difference between and ExternalAsset and a DownloadAsset? I don't get it from the javadoc

File: mzmine-community/src/main/java/io/github/mzmine/modules/io/download/DownloadAssetButton.java [hunk 4]
```diff
@@ -129,10 +146,47 @@ private ButtonBase addDownloadLinksButton(final List<DownloadAsset> assets) {
...
+    var menu = new CustomMenuItem(main);
+    menu.setOnAction(_ -> downloadAction.run());
+    return menu;
+  }
+
   /**
    * Starts a download task and blocks any further download attempts
    */
-  private void download(final DownloadAsset asset) {
+  private static void download(final DownloadAsset asset, final BooleanProperty isDownloading,
```
Comment (SteffenHeu) [inline mzmine-community/src/main/java/io/github/mzmine/modules/io/download/DownloadAssetButton.java:187]: i think something in the estimated file is not working, it always re-downloads the everything for me

File: mzmine-community/src/main/java/io/github/mzmine/modules/io/download/DownloadAssetButton.java [hunk 5]
```diff
@@ -129,10 +146,47 @@ private ButtonBase addDownloadLinksButton(final List<DownloadAsset> assets) {
...
+        FxIconUtil.newIconButton(FxIcons.X_CIRCLE, "Cancel download", EventHandling.CONSUME_EVENTS,
+            () -> cancelCurrentTask(taskProperty)));
+    HBox.setHgrow(progressBar, Priority.ALWAYS);
+    progressPane.setAlignment(Pos.CENTER);
+    progressPane.visibleProperty().bind(isDownloading);
+    progressPane.managedProperty().bind(isDownloading);
+    main.setTop(progressPane);
+    Runnable downloadAction = () -> {
+      if (!isDownloading.get()) {
+        download(asset, isDownloading, files -> onDownloadFinished.accept(files),
```
Comment (SteffenHeu) [inline mzmine-community/src/main/java/io/github/mzmine/modules/io/download/DownloadAssetButton.java:168]: if the download consists of multiple files like the msnlib, the progress only shows the progress of a single file, not all files and thus jumps around

**Review by SteffenHeu — COMMENTED** [review]
got some questions

File: mzmine-community/src/main/java/io/github/mzmine/modules/io/download/DownloadAssetButton.java (hunk 3 above)
Comment (robinschmid) [reply mzmine-community/src/main/java/io/github/mzmine/modules/io/download/DownloadAssetButton.java:72]: renamed ExternalAsset to AssetGroup

File: mzmine-community/src/main/java/io/github/mzmine/modules/io/download/DownloadAssetButton.java (hunk 4 above)
Comment (robinschmid) [reply mzmine-community/src/main/java/io/github/mzmine/modules/io/download/DownloadAssetButton.java]: redone

**Pull request merged**